ledger/benches/blockstore_processor.rs

```diff
@@ -0,0 +1,185 @@
+                executable: false,
+                rent_epoch: Epoch::MAX,
+            },
+        );
+    });
+
+    accounts
+}
+
+fn create_transactions(bank: &Bank, num: usize) -> Vec<SanitizedTransaction> {
+    let funded_accounts = create_funded_accounts(bank, 2 * num);
+    funded_accounts
+        .into_par_iter()
+        .chunks(2)
+        .map(|chunk| {
+            let from = &chunk[0];
+            let to = &chunk[1];
+            system_transaction::transfer(from, &to.pubkey(), 1, bank.last_blockhash())
+        })
+        .map(SanitizedTransaction::from_transaction_for_tests)
+        .collect()
+}
+
+struct BenchFrame {
+    bank: Arc<Bank>,
+    prioritization_fee_cache: PrioritizationFeeCache,
+}
+
+fn setup(apply_cost_tracker_during_replay: bool) -> BenchFrame {
+    let mint_total = u64::MAX;
+    let GenesisConfigInfo {
+        mut genesis_config, ..
+    } = create_genesis_config(mint_total);
+
+    // Set a high ticks_per_slot so we don't run out of ticks
+    // during the benchmark
+    genesis_config.ticks_per_slot = 10_000;
+
+    let mut bank = Bank::new_for_benches(&genesis_config);
+
+    if !apply_cost_tracker_during_replay {
+        bank.deactivate_feature(&apply_cost_tracker_during_replay::id());
+    }
+
+    // Allow arbitrary transaction processing time for the purposes of this bench
+    bank.ns_per_slot = u128::MAX;
+
+    // set cost tracker limits to MAX so it will not filter out TXs
+    bank.write_cost_tracker()
+        .unwrap()
+        .set_limits(std::u64::MAX, std::u64::MAX, std::u64::MAX);
+    let bank = bank.wrap_with_bank_forks_for_tests().0;
+    let prioritization_fee_cache = PrioritizationFeeCache::default();
+    BenchFrame {
+        bank,
+        prioritization_fee_cache,
+    }
+}
+
+fn bench_execute_batch(
+    bencher: &mut Bencher,
+    batch_size: usize,
+    apply_cost_tracker_during_replay: bool,
+) {
+    const TRANSACTIONS_PER_ITERATION: usize = 64;
+    assert_eq!(
+        TRANSACTIONS_PER_ITERATION % batch_size,
+        0,
+        "batch_size must be a factor of \
+        `TRANSACTIONS_PER_ITERATION` ({TRANSACTIONS_PER_ITERATION}) \
+        so that bench results are easily comparable"
+    );
+    let batches_per_iteration = TRANSACTIONS_PER_ITERATION / batch_size;
+
+    let BenchFrame {
+        bank,
+        prioritization_fee_cache,
+    } = setup(apply_cost_tracker_during_replay);
+    let transactions = create_transactions(&bank, 2_usize.pow(20));
+    let batches: Vec<_> = transactions
+        .chunks(batch_size)
+        .map(|txs| {
+            let mut batch =
+                TransactionBatch::new(vec![Ok(()); txs.len()], &bank, Cow::Borrowed(txs));
+            batch.set_needs_unlock(false);
+            TransactionBatchWithIndexes {
+                batch,
+                transaction_indexes: (0..batch_size).collect(),
+            }
+        })
+        .collect();
+    let mut batches_iter = batches.into_iter();
```

Comment:
use of `.into_iter()` here causes `::drop()`s are done inside `bencher.iter(...)`. seems the overall number isn't affected that. but this poped up in the profiler:

![image](https://github.com/solana-labs/solana/assets/117807/a374a595-8e4f-48e5-ac26-706fdf755ea3)


so, how about doing drop after `bencher.iter(...)` like this for perfection?

```patch
diff --git a/ledger/benches/blockstore_processor.rs b/ledger/benches/blockstore_processor.rs
index ffe644ac36..4617fe648f 100644
--- a/ledger/benches/blockstore_processor.rs
+++ b/ledger/benches/blockstore_processor.rs
@@ -135,14 +135,14 @@ fn bench_execute_batch(
             }
         })
         .collect();
-    let mut batches_iter = batches.into_iter();
+    let mut batches_iter = batches.iter();
 
     let mut timing = ExecuteTimings::default();
     bencher.iter(|| {
         for _ in 0..batches_per_iteration {
             let batch = batches_iter.next().unwrap();
             let _ = execute_batch(
-                &batch,
+                batch,
                 &bank,
                 None,
                 None,
@@ -152,6 +152,8 @@ fn bench_execute_batch(
             );
         }
     });
+    // make sure drop of batches aren't accounted for bench results
+    drop(batches);
 }

```

Reply:
2d36845